src/ansys/fluent/core/utils/search.py

```diff
@@ -212,10 +238,320 @@ def inner(tree, path, root_path):
+
+
+def _get_exact_match_for_word_from_names(
+    word: str,
+    names: list,
+):
+    """Get exact match for the given word.
+
+    Parameters
+    ----------
+    word: str
+        The word to search for.
+    names: list
+        All API object names.
+
+    Returns
+    -------
+    exact_matche: list
+        List of exact match.
+    """
+    return [name for name in names if word == name]
+
+
+def _get_close_matches_for_word_from_names(
+    word: str,
+    names: list,
+):
+    """Get close matches for the given word.
+
+    Parameters
+    ----------
+    word: str
+        The word to search for.
+    names: list
+        All API object names.
+
+    Returns
+    -------
+    valid_close_matches: list
+        List of valid close matches.
+    """
+    close_matches = closest_allowed_names(word, names)
+    valid_close_matches = [
+        close_match for close_match in close_matches if close_match in names
+    ]
+    return valid_close_matches
+
+
+def _search_whole_word(search_string: str, match_case: bool):
+    """Perform exact search for a word through the Fluent's object hierarchy.
+
+    Parameters
+    ----------
+    search_string: str
+        The word to search for. Semantic search is default.
+    match_case: bool
+        Whether to match case. If ``True`` will match case-insensitive case.
+
+    Returns
+    -------
+        List of search string matches.
+    """
+    names = _get_api_object_names()
+    if match_case:
+        search_string = search_string + "*"
+        _search_wildcard(search_string, names)
+    else:
+        queries = _get_exact_match_for_word_from_names(search_string, names)
+        if queries:
+            _print_search_results(queries, wildcard=False)
+
+
+def _download_nltk_data():
+    """Download NLTK data on-demand."""
+    import nltk
+
+    packages = ["wordnet", "omw-1.4"]
+    for package in packages:
+        nltk.download(package, quiet=True)
+
+
+def _search_semantic(search_string: str, language: str):
+    """Perform semantic search for a word through the Fluent's object hierarchy.
+
+    Parameters
+    ----------
+    search_string: str
+        The word to search for. Semantic search is default.
+    language: str
+        The language for the semantic search.
+        English is default for the semantic search.
+        ISO 639-3 code of the language to be used for semantic search.
+        See `https://omwn.org/omw1.html` for the list of supported languages.
+        The default value is `eng` for English language.
+
+    Returns
+    -------
+    queries: list
+        List of search string matches.
+    """
+    names = _get_api_object_names()
+    similar_keys = set()
+    search_string_synsets = (
+        wn.synsets(search_string.decode("utf-8"), lang=language)
+        if sys.version_info[0] < 3
+        else wn.synsets(search_string, lang=language)
+    )
+    for name in names:
+        api_object_name_synsets = (
```

Comment:
Is `api_object_name_synsets` always in English (because api_names are in English)? We can store this result (the dictionary of name to synsets or the reverse dictionary of synonym to names) in the file containing the names. So, this result will be pre-computed.

Reply:
Is api_object_name_synsets always in English (because api_names are in English)? - yes

Reply:
That's right, it will improve the performance. Thank you.

Reply:
https://github.com/ansys/pyfluent/issues/2844